R/MsBackendTimsTof-functions.R

```diff
@@ -177,3 +175,52 @@ MsBackendTimsTof <- function() {
     xn[grep("^(n|-)", x, ignore.case = TRUE)] <- 0L
     xn
 }
+
+# can we assume that tms@all_coulmns is the same for all the TimsTOF? 
+.TIMSTOF_COLUMNS <- c("mz", "intensity", "tof", "inv_ion_mobility")
+
+.spectra_data <- function(x, columns = spectraVariables(x)) {
+  if (!all(present <- columns %in% spectraVariables(x)))
+    stop("Column(s) ", paste0("\"", columns[!present], "\"", collapse = ", "),
+         " not available.", call. = FALSE)
+  res <- vector(mode = "list", length = length(columns))
+  names(res) <- columns
+  core_cols <- columns[columns %in% names(Spectra:::.SPECTRA_DATA_COLUMNS)]
```

Comment:
@jorainer Should we export the `.SPECTRA_DATA_COLUMNS` in `Spectra`?

Reply:
yes, that might be a good idea - but I guess it's better to not directly export the variable (what happens if a user changes this variable?). Maybe we could export a `coreSpectraVariables` function that returns it?